### ISSUE TITLE
Weird Exception that I can't seem to work out what I have done

### ISSUE THREAD

**Issue by MichaelPine** [mccalltd/AttributeRouting#133]
I have previously used attribute routing, but I am now getting the following exception with this current aplication I am working. However, please not this was working, I have uninstalled and installed the package about 3 times now, so that ain't fixing it. here is the exception.

I really have not done anything that I am "aware" of, so was wondering if someone might know of something I need to clear out, I was running this locally and debugging no worries.

Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information. 
Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. 

Exception Details: System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.

Source Error: 

Line 11:            // To debug routes locally using the built in ASP.NET development server, go to /routes.axd
Line 12:  
Line 13:            routes.MapAttributeRoutes();
Line 14:        }
Line 15:  

Source File: C:\Assets_CapitalWorks\Trunk\CapitalWorks\CapitalWorks\App_Start\AttributeRouting.cs    Line: 13 

Stack Trace: 

[ReflectionTypeLoadException: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.]
   System.Reflection.RuntimeModule.GetTypes(RuntimeModule module) +0
   System.Reflection.RuntimeModule.GetTypes() +4
   System.Reflection.Assembly.GetTypes() +61
   AttributeRouting.Helpers.ReflectionExtensions.GetControllerTypes(Assembly assembly, Type controllerType) +97
   AttributeRouting.AttributeRoutingConfigurationBase.AddRoutesFromAssembly(Assembly assembly) +90
   AttributeRouting.Web.Mvc.RouteCollectionExtensions.MapAttributeRoutes(RouteCollection routes) +94
   CapitalWorks.App_Start.AttributeRouting.RegisterRoutes(RouteCollection routes) in C:\Assets_CapitalWorks\Trunk\CapitalWorks\CapitalWorks\App_Start\AttributeRouting.cs:13
   CapitalWorks.App_Start.AttributeRouting.Start() in C:\Assets_CapitalWorks\Trunk\CapitalWorks\CapitalWorks\App_Start\AttributeRouting.cs:17

[TargetInvocationException: Exception has been thrown by the target of an invocation.]
   System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor) +0
   System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments) +192
   System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) +108
   System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters) +19
   WebActivator.PreApplicationStartMethodAttribute.InvokeMethod() in D:\Code\Bitbucket\WebActivator\WebActivator\PreApplicationStartMethodAttribute.cs:42
   WebActivator.PreApplicationStartCode.Start() in D:\Code\Bitbucket\WebActivator\WebActivator\PreApplicationStartCode.cs:20

[InvalidOperationException: The pre-application start initialization method Start on type WebActivator.PreApplicationStartCode threw an exception with the following error message: Exception has been thrown by the target of an invocation..]
   System.Web.Compilation.BuildManager.InvokePreStartInitMethodsCore(ICollection`1 methods, Func`1 setHostingEnvironmentCultures) +550
   System.Web.Compilation.BuildManager.InvokePreStartInitMethods(ICollection`1 methods) +132
   System.Web.Compilation.BuildManager.CallPreStartInitMethods(String preStartInitListPath) +90
   System.Web.Compilation.BuildManager.ExecutePreAppStart() +135
   System.Web.Hosting.HostingEnvironment.Initialize(ApplicationManager appManager, IApplicationHost appHost, IConfigMapPathFactory configMapPathFactory, HostingEnvironmentParameters hostingParameters, PolicyLevel policyLevel, Exception appDomainCreationException) +516

[HttpException (0x80004005): The pre-application start initialization method Start on type WebActivator.PreApplicationStartCode threw an exception with the following error message: Exception has been thrown by the target of an invocation..]
   System.Web.HttpRuntime.FirstRequestInit(HttpContext context) +9873912
   System.Web.HttpRuntime.EnsureFirstRequestInit(HttpContext context) +101
   System.Web.HttpRuntime.ProcessRequestInternal(HttpWorkerRequest wr) +456

**Comment by Valamas** [comment]
When I had this it was during a VS2010 to VS20103 + MVC3 to MVC5 upgrade. Not an easy task but it was completed successfully.

One problem I had was there were references to assemblies that were not showing up in project references. I found that the project.csproj still listed the references for older assemblies either that they were meant to be just removed or removed and a higher version added back. In the later case the older version number was being used instead of the newer version number which resulted in references to assemblies that did not exist anymore.

I had via NUGET and via manual deleting from references the following.
Removed System.Web.Helpers. But still in the csprof file. Wanted it removed completely. 
Removed AttributeRouting 3.1.0 and added 3.5.6. However, 3.1.0 was still listed in the csproj, so it was never removed in the earlier remove action. And the install of 3.5.6 did not touch the non-removed reference.

I then had to remove 3.5.6 and the reference from the csproj manually.
I then installed 3.5.6 again and it worked.

Hope this adds a little more info. There could be inaccuracies in my story due to the chaotic upgrade.

In summary, check the references in the proj file are correct.
